x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -786,7 +786,7 @@ protected LogicalPlan rule(OrderBy orderBy) {
 
             if (child instanceof OrderBy childOrder) {
                 // combine orders
-                return new OrderBy(orderBy.source(), childOrder.child(), CollectionUtils.combine(orderBy.order(), childOrder.order()));
+                return new OrderBy(orderBy.source(), childOrder.child(), orderBy.order());
```

Comment:
Another way to write the above is to do:
`return orderBy.replaceChild(childOrder.child)`

No need to update the PR for it.